packages/data-table-server/src/dataTableService/DataTableServiceBuilder.ts

```diff
@@ -11,6 +11,7 @@ import {
   EntityRelationsDataTableService,
   EventsDataTableService,
 } from './internal';
+import { SqlDataTableService } from './userDefined';
 
 /**
  * Generic builder class that allows us to configure the context for a specific DataTableService
```

Comment:
DataTableServiceBuilder could just be a fn? I don't see the need for OOP here?

Reply:
That is true. Tbh, the main reason I went with the builder pattern here is to get the generics working how I wanted it to.

With the current implementation, you get some pretty neat behaviour that ensures that the `context` that gets passed in meets the requirements of the `serviceType`, ie.
```
new DataTableServiceBuilder()
  .setServiceType('entities')
  .setContext({ models: tupaiaModels }) // ts error: models is not assignable to parameter of type EntitiesDataTableContext
  .build();
```

I've tried my hand at getting this working in a function, but couldn't quite work it out. Did you have any suggestions on how to get this working, I'd be down to pair on it for a bit 😄 

If it's impossible tho, I'd prefer to keep the functionality as is, since the type validation on the context object is really clean atm 👍

Reply:
Hmm yeah I see what you mean. My first thought would be to just have the same context passed in to each service.

But if we want each service to have a subset of the context, I think something like this would work? 
https://github.com/beyondessential/tupaia/commit/a9b0f8612ee68e1de5e3c75ae2a25ec7fb1b04c4

Reply:
Discussed, builder is cool 😎